src/lib/transform.ts

```diff
@@ -1,30 +1,16 @@
-import { alphabetize, has } from '../utils/object';
-
-export interface Transformation {
-  (key: string, prevValue: any): [string, string];
-}
-export interface Transformations {
-  [key: string]: Transformation;
-}
+import { Transformations, PackageJson } from '../types';
 
 // All transformers receive:
 //   * the key they matched on
-//   * the value in package.json (if any)
+//   * the value in package.PackageJson (if any)
 //
 // Return a new key and value to be stored
 export default async function transform(
   prevKey: string,
-  prevValue: any,
-  transformations: Transformations = {}
-): Promise<[string, any]> {
-  let nextKey = prevKey;
-  let nextValue = prevValue;
-
-  if (has(transformations, prevKey)) {
-    [nextKey, nextValue] = await transformations[prevKey](prevKey, prevValue);
-  } else if (typeof prevValue === 'object') {
-    nextValue = await alphabetize(prevValue);
-  }
-
-  return [nextKey, nextValue];
+  prevValue: PackageJson,
+  transformations: Transformations
+): Promise<[string, PackageJson]> {
+  // If there are no transformations, use the default transform
+  const transformation = transformations[prevKey] || transformations['*'];
```

Comment:
You're using null coalescing operator in multiple places. Here, a null propagating could also do the job, and let pass through the eventual addition of `false`.